File: components/mutations/RefreshFeed.ts
import { Environment,graphql } from "react-relay"

import { commitMutationAsync } from "./commitMutationAsync"

const mutation = graphql`
  mutation RefreshFeedMutation($input: RefreshFeedInput!) {
    refreshFeed(input: $input) {
      feed {
        id
        title
        homePageURL
        refreshedAt
      }
    }
  }
`

export async function refreshFeed(
  environment: Environment,
  id: string
): Promise<void> {
  const variables = { input: { id } }
  await commitMutationAsync(environment, {
    mutation,
    variables,
  })
}

File: ios/Courier/Courier/Mutations/RefreshFeed.swift
import RelaySwiftUI
import CourierGenerated

private let mutation = graphql("""
mutation RefreshFeedMutation($input: RefreshFeedInput!) {
  refreshFeed(input: $input) {
    feed {
      id
      refreshing
    }
  }
}
""")

extension Mutation.Mutator where Operation == RefreshFeedMutation {
    func commit(id: String) {
        commit(
            variables: .init(input: .init(id: id)),
            optimisticResponse: [
                "refreshFeed": [
                    "feed": [
                        "id": id,
                        "refreshing": true,
                    ]
                ]
            ]
        )
    }
}
